[Yap Zong Xin] iP

yap-zong-xin opened the pull request:
### DukePro
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ *SUPER* FAST to use

All you need to do is,
1. download it from [here](https://nus-cs2103-ay2324s2.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

---

If you are a Java programmer, you can use it to practice Java too. Here's the `main`  method:
```ruby
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

yap-zong-xin committed:
Level-0

yap-zong-xin committed:
Level-1

yap-zong-xin committed:
Level-2

yap-zong-xin committed:
Level-3

yap-zong-xin committed:
Level-4

yap-zong-xin committed:
A-TextUiTesting

yap-zong-xin committed:
Level-5

yap-zong-xin committed:
Level-6

yap-zong-xin committed:
Update relevant java files for Level-7

yap-zong-xin committed:
Update Duke.java and duck.txt files

jaejayrome commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,177 @@
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+    private static final String FILE_PATH = "./data/duke.txt";
```
I like how you have adhered to the naming of constants!

jaejayrome commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,177 @@
+        System.out.println("    Hello! I'm Yappy\n    What can I do for you?");
+        System.out.println("    ____________________________________________________________");
+
+        ArrayList<Task> AL = new ArrayList<>();
```
It would have been better if you could change the name of this arraylist to 'taskList' / 'tasks' to made it easier for others to know what is this arraylist needed for :)

jaejayrome commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,177 @@
+
+            Task task;
+
+            switch (type) {
+                case "T":
+                    task = new Todos(description);
+                    break;
+                case "D":
+                    if (parts.length >= 4) {
+                        String by = parts[3];
+                        task = new Deadline(description, by);
+                    } else {
+                        task = new Task(description);
+                    }
+                    break;
+                case "E":
+                    if (parts.length >= 5) {
+                        String from = parts[3];
+                        String to = parts[4];
+                        task = new Events(description, from, to);
+                    } else {
+                        task = new Task(description);
+                    }
+                    break;
+                default:
+                    task = new Task(description);
+            }
+
```
It would be better if you have avoided 'deep nesting' within your methods and also the indentation for the switch cases should be on the same as the opening switch statement :))

jaejayrome commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,177 @@
+        loadTasksFromFile(AL, FILE_PATH);
+
+        while(true) {
+            BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
```
Just a small note: I feel that you could adhere to the use of 'bufferedReader' instead of 'br' because later on in the code 'bufferedWriter' was used to name the BufferedWriter instance; just to keep it consistent

jaejayrome commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,177 @@
+                System.out.println("       " + AL.get(Integer.parseInt(token[1]) - 1));
+                System.out.println("    ____________________________________________________________");
+            } else if (token[0].equals("deadline")) {
+//                Task[] tasks = new Task[100];
```
This comment could be removed since you are already using an arraylist to store your tasks :)

jaejayrome commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,29 @@
+  }
+
+  public String getStatusIcon() {
+    return (isDone ? "X" : " "); // mark done task with X
```
This comment could be placed above the return statement as well :)

jaejayrome reviewed (commented):
LGTM! Just have some minor nits that's all, keep up the good work 🥇

yap-zong-xin committed:
add files for branch-level-8

yap-zong-xin committed:
Update files for branch-A-MoreOOP

yap-zong-xin committed:
Move all java files to duke package

yap-zong-xin committed:
Merge branch 'add-gradle-support'

yap-zong-xin committed:
add gradle

yap-zong-xin committed:
add gradle

yap-zong-xin committed:
add 2 test files

yap-zong-xin committed:
add A-Jar

yap-zong-xin committed:
update to include JavaDoc

yap-zong-xin committed:
update files to comply with coding standards

yap-zong-xin committed:
update files to comply to more coding standards

yap-zong-xin committed:
update files to resolve merge conflicts

yap-zong-xin committed:
add coding standards

yap-zong-xin committed:
add find method

yap-zong-xin committed:
add gui features

yap-zong-xin committed:
Various classes and methods lack assertions, potentially leading to runtime errors and unexpected behavior.
Hence assertion is needed to enhance code reliability and maintainability by enforcing preconditions and improving error handling.
Assertions are being added to constructors and methods to validate inputs and handle errors more effectively. Null checks and input validation are included to ensure robustness. Additionally, token array validity and task list integrity are verified in the TaskList class.
By incorporating assertions, we create guardrails that catch potential issues early, allowing for faster debugging and more predictable behavior.
This change improves code quality and readability, making the system more resilient to unexpected inputs and conditions.

yap-zong-xin committed:
Improve code quality
Code readability improved, prevent unsafe practice, and added necessary code comments

Deep nesting, long lines, complicated expressions, and lack of clarity in the happy path hinder readability and maintainability.

Let's,
* Remove deep nesting.
* Ensure each line is not too long.
* Avoid complicated expressions.
* Make the happy path more clear and prominent.
* Prevent unsafe practices.
* Add necessary code comments.

These enhancements improve code readability, aiding comprehension, maintenance, and debugging. By reducing deep nesting and avoiding complex expressions, the code's complexity is minimized. Emphasizing the happy path clarifies the main intent, while preventing unsafe practices ensures reliability and security. Furthermore, adding relevant comments enhances documentation for easier maintenance and future modifications.

Consistently applying these improvements enhances the overall quality of the codebase and fosters a more maintainable and scalable development process.

yap-zong-xin committed:
Add BCD-Extension

yap-zong-xin committed:
Merge pull request #2 from yap-zong-xin/branch-BCD-Extension

Add BCD-Extension

yap-zong-xin committed:
Add increments for A-BetterGui

yap-zong-xin committed:
Add increment for A-Personality

yap-zong-xin committed:
Add increment for A-MoreErrorHandling

yap-zong-xin committed:
Remove unused warning image

yap-zong-xin committed:
Remove line containing warning image import

yap-zong-xin committed:
Add Ui to docs

yap-zong-xin committed:
Add READMEmd file

yap-zong-xin committed:
Add A-Release by configuring build.gradle file
